pkg/blockstorage/azure/auth.go

```diff
@@ -30,28 +30,31 @@ type AzureAuthenticator interface {
 	Authenticate(creds map[string]string) error
 }
 
-func NewAzureAutheticator(config map[string]string) (AzureAuthenticator, error) {
+func NewAzureAuthenticator(config map[string]string) (AzureAuthenticator, error) {
 	switch {
 	case isMSICredsAvailable(config):
-		return &msiAuthenticator{}, nil
+		return &MsiAuthenticator{}, nil
 	case isClientCredsAvailable(config):
-		return &clientSecretAuthenticator{}, nil
+		return &ClientSecretAuthenticator{}, nil
 	default:
 		return nil, errors.New("Fail to get an authenticator for provided creds combination")
 	}
 }
 
 // authenticate with MSI creds
-type msiAuthenticator struct{}
+type MsiAuthenticator struct{}
 
-func (m *msiAuthenticator) Authenticate(creds map[string]string) error {
+// default-id: empty but existing clientID ?
```

Comment:
Is this question for me? :P

Reply:
oh no haha, it was question for myself on design points. Let me clean up